src/nix-build/nix-build.cc

```diff
@@ -271,10 +271,14 @@ void mainWrapped(int argc, char * * argv)
         exprs = {state.parseStdin()};
     else
         for (auto i : left) {
+            auto absolute = i;
+            try {
+                absolute = canonPath(absPath(i), true);
```

Comment:
Do we care that this will also resolve symlinks in the store?

Reply:
I don't think so, since this only affects the .drv case.

Reply:
And what happens if there are symlinks in the store path (i.e. `/nix/` is a symlink)? Does it move from «purity not guaranteed» to «doesn't work»?

Reply:
/nix is not allowed to be a symlink anyway